tests/requests.py

```diff
@@ -1,90 +1,140 @@
+            validation_error.get("error_thrown").get("message"),
+            "'response' is a required property",
         )
 
-    def test_missing_response(self):
-        body = {"answer": "example", "params": {}}
-        validation_error = validate_request(body)
+    def test_null_response_for_eval(self):
+        body = {"response": None, "answer": "example", "params": {}}
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
 
         self.assertNotEqual(validation_error, None)
 
         self.assertEqual(
             validation_error.get("error_thrown").get("message"),
-            "'response' is a required property") 
+            "None should not be valid under {'type': 'null'}",
+        )
 
-    def test_null_response(self):
+    def test_null_response_for_preview(self):
         body = {"response": None, "answer": "example", "params": {}}
-        validation_error = validate_request(body)
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
+
+        self.assertNotEqual(validation_error, None)
+
+        self.assertEqual(
+            validation_error.get("error_thrown").get("message"),
+            "None should not be valid under {'type': 'null'}",
+        )
+
+    def test_missing_answer_in_eval(self):
+        body = {"response": "example", "params": {}}
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
 
         self.assertNotEqual(validation_error, None)
 
         self.assertEqual(
             validation_error.get("error_thrown").get("message"),
-            "None should not be valid under {'type': 'null'}")
-    
-    def test_missing_answer(self):
+            "'answer' is a required property",
+        )
+
+    def test_missing_answer_in_preview(self):
         body = {"response": "example", "params": {}}
-        validation_error = validate_request(body)
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
 
         self.assertNotEqual(validation_error, None)
 
         self.assertEqual(
             validation_error.get("error_thrown").get("message"),
-            "'answer' is a required property")  
-     
+            "'answer' is a required property",
+        )
+
+    def test_including_answer_in_eval(self):
+        body = {"response": "example", "params": {}}
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
+
+        self.assertNotEqual(validation_error, None)
+
+        self.assertEqual(
+            validation_error.get("error_thrown").get("message"),
+            "'answer' is a required property",
+        )
+
+    def test_including_answer_in_preview(self):
+        body = {"response": "example", "params": {}}
+        validation_error = validate.body(body, ReqBodyValidators.GENERIC)
+
+        self.assertNotEqual(validation_error, None)
+
+        self.assertEqual(
+            validation_error.get("error_thrown").get("message"),
+            "'answer' is a required property",
+        )
```

Comment:
What's the difference between all these tests?

Reply:
These test the different commands, which use separate schemas. Although they are checking the same field (e.g. answer), each has different requirements: eval must have an answer field, but preview must not.

Reply:
NB this change is outdated from an older commit